Lib/multiprocessing/resource_tracker.py

```diff
@@ -159,7 +178,17 @@ def unregister(self, name, rtype):
         self._send('UNREGISTER', name, rtype)
 
     def _send(self, cmd, name, rtype):
-        self.ensure_running()
+        try:
+            self.ensure_running()
+        except ReentrantCallError:
```

Comment:
I don't get this code path. So it's possible that self._fd is None, since something is calling ensure_running(), no? But the code below doesn't check if self._fd is None or not.

Reply:
This is true, but `self._fd` could be non-None but point to a dead process. See `_check_alive()`.